drivers/wifi/nrf700x/CMakeLists.txt

```diff
@@ -129,14 +129,61 @@ zephyr_compile_definitions_ifdef(CONFIG_NRF700X_ON_QSPI
 )
 
 # RPU FW patch binaries based on the selected configuration
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SYSTEM_MODE
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin
-)
+if(CONFIG_NRF700X_SYSTEM_MODE)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/default/nrf70.bin)
+elseif (CONFIG_NRF700X_RADIO_TEST)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin)
+elseif (CONFIG_NRF700X_SCAN_ONLY)
+	set (NRF70_PATCH ${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin)
+else()
+	# Error
+	message(FATAL_ERROR "Unsupported nRF70 patch configuration")
+endif()
 
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_RADIO_TEST
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/radio_test/nrf70.bin
+zephyr_compile_definitions(
+	-DCONFIG_NRF_WIFI_FW_BIN=${NRF70_PATCH}
 )
 
-zephyr_compile_definitions_ifdef(CONFIG_NRF700X_SCAN_ONLY
-	-DCONFIG_NRF_WIFI_FW_BIN=${OS_AGNOSTIC_BASE}/fw_bins/scan_only/nrf70.bin
-)
+if (CONFIG_NRF_WIFI_PATCHES_EXT_FLASH_STORE)
+message(STATUS "nRF WiFi FW patch binary will be stored in external flash")
+
+if (CONFIG_BOARD_NRF5340DK_NRF5340_CPUAPP)
+set(NRF_QSPI_FLASH_BASE 0x10000000)
+elseif(CONFIG_BOARD_NRF52840DK_NRF52840)
+set(NRF_QSPI_FLASH_BASE 0x12000000)
+else()
+message(FATAL_ERROR "Unsupported board: ${BOARD}")
+endif()
+
+dt_nodelabel(nrf70_fw_node NODELABEL "nrf70_fw_partition")
+dt_prop(nrf70_fw_offset PATH ${nrf70_fw_node} PROPERTY reg INDEX 0)
+if (NOT DEFINED nrf70_fw_offset)
+	message(FATAL_ERROR "nrf70_fw_partition.reg not found in dts")
+endif()
+message(STATUS "nrf70_fw_offset: ${nrf70_fw_offset}")
+math(EXPR NRF70_PATCH_ADDR "${NRF_QSPI_FLASH_BASE} + ${nrf70_fw_offset}")
```

Comment:
this is mixing PM and dts, this is a no

Reply:
There is no PM involved here, instead we do this by manually using DTS. `NRF_QSPI_FLASH_BASE` is just a variable in this file.

Reply:
And that's the problem, ncs uses PM for everything, this goes completely opposite and reverts back to dts whilst the application continues to use PM - so how does this work when someone wants firmware updates for the network core store in QSPI?

Reply:
For now this feature is supported only for samples that do not use PM, Integrating with PM + DFU support is scheduled next with the help from Matter team, so, at that time, the `NRF70_PATCH_ADDR` will be derived from PM which should solve the multiple users storing in external memory issue, if PM isn't enabled then the current code will be used.

Reply:
Then how does this work? That means there can be only the application itself, no bootloader, no network core, no other image

Reply:
Yes, for Wi-Fi samples with this feature enabled that is fine, if any of them need to be used then this feature won't work for now, we are doing this in a phased manner. If you want I cann add `depends on !PM_ENABLED`?